tests/core/test_globaldns.py

```diff
@@ -1,29 +1,64 @@
 from .common import random_str
 from rancher import ApiError
 import pytest
+import time
+import kubernetes
 
 
-def test_dns_fqdn_unique(admin_mc):
+def test_dns_fqdn_unique(admin_mc, user_factory, remove_resource):
     client = admin_mc.client
     provider_name = random_str()
     access = random_str()
     secret = random_str()
+    # Add regular user as member to gdns provider
+    user_member = user_factory()
+    remove_resource(user_member)
+    user_client = user_member.client
+    members = [{"userPrincipalId": "local://" + user_member.user.id,
+                "accessType": "owner"}]
     globaldns_provider = \
         client.create_global_dns_provider(
                                          name=provider_name,
                                          route53ProviderConfig={
                                              'accessKey': access,
                                              'secretKey': secret,
-                                             'rootDomain': "example.com"})
+                                             'rootDomain': "example.com"},
+                                         members=members)
 
     fqdn = random_str() + ".example.com"
     globaldns_entry = \
-        client.create_global_dns(fqdn=fqdn, providerId=provider_name)
+        client.create_global_dns(fqdn=fqdn, providerId=provider_name,
+                                 members=members)
+
+    # Make sure creator can access both, provider and entry
+    gdns_provider_id = "cattle-global-data:" + provider_name
+    gdns_provider = client.by_id_global_dns_provider(gdns_provider_id)
+    assert gdns_provider is not None
+
+    gdns_entry_id = "cattle-global-data:" + globaldns_entry.name
+    gdns = client.by_id_global_dns(gdns_entry_id)
+    assert gdns is not None
 
     with pytest.raises(ApiError) as e:
         client.create_global_dns(fqdn=fqdn, providerId=provider_name)
         assert e.value.error.status == 422
 
+    # user should be able to list this gdns provider
+    api_instance = kubernetes.client.RbacAuthorizationV1Api(
+        admin_mc.k8s_client)
+    provider_rb_name = provider_name + "-gp-a"
+    wait_to_ensure_user_in_rb_subject(api_instance, provider_rb_name,
+                                      user_member.user.id)
+    gdns_provider = user_client.by_id_global_dns_provider(gdns_provider_id)
+    assert gdns_provider is not None
+
+    # user should be able to list this gdns entry
+    entry_rb_name = globaldns_entry.name + "-g-a"
+    wait_to_ensure_user_in_rb_subject(api_instance, entry_rb_name,
+                                      user_member.user.id)
+    gdns = user_client.by_id_global_dns(gdns_entry_id)
+    assert gdns is not None
+
```

Comment:
Can we keep the two tests separate - the one that checks fqdn unique - and new one to check access for listing